zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -73,6 +84,10 @@
   private static Logger LOG = LoggerFactory.getLogger(ZkClient.class);
   private static long MAX_RECONNECT_INTERVAL_MS = 30000; // 30 seconds
 
+  // Default value for system property jute.maxbuffer
+  // It specifies the maximum size of the data that can be stored in a znode.
+  private static final int DEFAULT_JUTE_MAXBUFFER = 0xfffff;
```

Comment:
I'm not sure if we can directly use the value ClientCnxn.packetLen. It is public and static.

Reply:
Answer is no. That one is different: it's for the response packet size and it's 4 MB. We are looking at server side jute.maxbuffer that is by default 1 MB in `BinaryInputArchive`. It does no harm to define our own private constant and the default value 1 MB is known.

Reply:
What if ZK changes the default max buffer in some new versions? Are we going to change this number accordingly? So my suggestion is that,
1. We fine the default value in Zookeeper code.
2. if 1 is not possible, then we can just use the current SIZE_LIMIT for the default value.

Reply:
Then you run into the ZK server side buffer issue. The write might be done successfully, but the read will fail.
I think we agreed to keep the buffer until we bump up the ZK version to a newer one and the buffer issue should be fixed in that version.

Reply:
@jiajunwang For default value 1000KB in ZNRecord SIZE_LIMIT, we still leave a margin that is 24 KB. But for the value that users config `jute.maxbuffer`, we should honor it as what is done in ZNRecordSerializer. We don't leave a margin in ZNRecordSerializer, right? So if user configs jute.maxbuffer, we expect them to understand it well. Say they already consider a margin in the value jute.maxbuffer, we should not(users don't expect that, either) leave another redundant margin. 

Say a user sets `jute.maxbuff` to 975KB that they think it already leaves enough room as margin, we don't want to deduct it again 975 - 24 = 951KB as the real value. It would also surprise user: if data is 960KB that user expects it to be written successfully, but it actually fails because of the extra margin.

So for the custom value if a user sets`jute.maxbuff`, I don't think we should deduct the value with the 24KB margin.